tasks/configure.yml

```diff
@@ -24,8 +33,17 @@
     path: "{{ etc_prefix }}/haproxy/backends.d"
     state: directory
 
-- name: 'Empty the folder if not already empty'
-  command: find {{ etc_prefix }}/haproxy/backends.d -name *.cfg -exec rm -f {} \;
+- name: "List files for the backends"
+  shell: ls -1 "{{ etc_prefix }}/haproxy/backends.d"
+  register: directory_contents
+  changed_when: false
+
+- name: "Remove unmanaged files for the backends"
+  file:
+    path: "{{ etc_prefix }}/haproxy/backends.d/{{item}}"
+    state: absent
+  when: item not in haproxy_backends|json_query('[*].name')|map('regex_replace',  '(.*)', '\\1.cfg')|list
```

Comment:
`json_query` was added in Ansible 2.2, so please bump the minimum version in `meta/main.yml`.